src/Symfony/Component/ImportMaps/ImportMapManager.php

```diff
@@ -0,0 +1,276 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\ImportMaps;
+
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Component\HttpClient\ScopingHttpClient;
+use Symfony\Component\VarExporter\VarExporter;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @experimental
+ *
+ * @author Kévin Dunglas <kevin@dunglas.dev>
+ */
+final class ImportMapManager
+{
+    public const POLYFILL_URL = 'https://ga.jspm.io/npm:es-module-shims@1.7.0/dist/es-module-shims.js';
+
+    /**
+     * @see https://regex101.com/r/2cR9Rh/1
+     *
+     * Partially based on https://github.com/dword-design/package-name-regex
+     */
+    private const PACKAGE_PATTERN = '/^(?:https?:\/\/[\w\.-]+\/)?(?:(?<registry>\w+):)?(?<package>(?:@[a-z0-9-~][a-z0-9-._~]*\/)?[a-z0-9-~][a-z0-9-._~]*)(?:@(?<version>[\w\._-]+))?(?:(?<subpath>\/.*))?$/';
+
+    private HttpClientInterface $apiHttpClient;
+    private ?array $importMap = null;
+
+    public function __construct(
+        private readonly string $path = 'importmap.php',
+        private readonly string $assetsDir = 'assets/',
+        private readonly string $publicAssetsDir = 'public/assets/',
+        private readonly string $assetsUrl = '/assets/',
+        private readonly Provider $provider = Provider::Jspm,
+        private ?HttpClientInterface $httpClient = null,
+        private readonly string $api = 'https://api.jspm.io',
```

Comment:
This one must allow downloading files from random domains because of the "download" features. (the JS can be hosted on any provider). But let's not use a scoped client then, it's simpler.